AndroidTest/app/src/main/java/cloudant/com/androidtest/keyprovider/KeyStorageTests.java

```diff
@@ -0,0 +1,104 @@
+package cloudant.com.androidtest.keyprovider;
+
+import android.test.AndroidTestCase;
+
+import com.cloudant.sync.datastore.encryption.KeyData;
+import com.cloudant.sync.datastore.encryption.KeyStorage;
+
+import java.util.Arrays;
+
+public class KeyStorageTests extends AndroidTestCase {
+
+    // Green Path Tests
+    public void testKeyDataSaveAndRetreive() {
+        KeyStorage storage = new KeyStorage(getContext(), ProviderTestUtil.getUniqueIdentifier());
+        KeyData original = ProviderTestUtil.createKeyData();
+
+        boolean saveSuccess = storage.saveEncryptionKeyData(original);
+        assertTrue("saveEncryptionKeyData failed", saveSuccess);
+
+        KeyData savedData = storage.getEncryptionKeyData();
+
+        assertTrue("saved KeyData should be equal to original", keyDataEquals(original, savedData));
+    }
+
+    public void testKeyDataExists() {
+        KeyStorage storage = new KeyStorage(getContext(), ProviderTestUtil.getUniqueIdentifier());
+        KeyData original = ProviderTestUtil.createKeyData();
+
+        boolean saveSuccess = storage.saveEncryptionKeyData(original);
+        assertTrue("saveEncryptionKeyData failed", saveSuccess);
+
+        boolean keyDataExists = storage.encryptionKeyDataExists();
+
+        assertTrue("KeyData should exist", keyDataExists);
```

Comment:
Changes requested in comment included in https://github.com/cloudant/sync-android/commit/dd276d263bfc9c5ddebb047cecd27230cb074f63